components/GameScreen.vue

```diff
@@ -56,6 +63,80 @@ type DrawBotArgs = {
   graphics: Graphics;
 };
 
+function smoothZoom(Pos: { x: number; y: number }, scale: number) {
+  if (!appRef.value) {
+    return;
+  }
+
+  if (isZooming) {
+    return; // Prevent multiple simultaneous zooms
+  }
+  isZooming = true;
+  targetScale = scale;
+  targetPos = Pos;
+  startZoomTime = performance.now();
+
+  requestAnimationFrame(animateZoom);
+}
+
+function animateZoom(currentTime: number) {
+  if (!appRef.value || !appRef.value.stage || !startZoomTime) {
+    return;
+  }
+
+  const elapsedTime = currentTime - startZoomTime;
+  const progress = Math.min(elapsedTime / zoomDuration, 1);
+
+  // Get the current scale and the scale change based on progress
+  const currentScale = appRef.value.stage.scale.x;
+  const newScale = currentScale + (targetScale - currentScale) * progress;
+
+  // Calculate the world position relative to the current scale
+  const worldPos = {
+    x: (targetPos.x - appRef.value.stage.position.x) / appRef.value.stage.scale.x,
+    y: (targetPos.y - appRef.value.stage.position.y) / appRef.value.stage.scale.y,
+  };
+
+  appRef.value.stage.scale.set(newScale);
+
+  const newScreenPos = {
+    x: worldPos.x * newScale + appRef.value.stage.position.x,
+    y: worldPos.y * newScale + appRef.value.stage.position.y,
+  };
+
+  // Adjust the stage position to follow the zoom focus point
+  appRef.value.stage.position.set(
+    Math.min(0, Math.max(appRef.value.stage.position.x - (newScreenPos.x - targetPos.x), appRef.value.screen.width - appRef.value.screen.width * newScale)),
+    Math.min(0, Math.max(appRef.value.stage.position.y - (newScreenPos.y - targetPos.y), appRef.value.screen.height - appRef.value.screen.height * newScale)),
+  );
+
+  if (progress < 1) {
+    requestAnimationFrame(animateZoom);
+  } else {
+    isZooming = false;
+  }
+}
+
+function followPlayerBot(botx: number, boty: number, username: string) {
+  if (!appRef.value || !gameState.value) {
+    return;
+  }
+  if (user.value?.body.username == username) {
+    const currentPos = appRef.value.stage.position;
+    const targetPos = {
+      x: -botx * appRef.value.stage.scale.x + appRef.value.screen.width / 2,
+      y: -boty * appRef.value.stage.scale.y + appRef.value.screen.height / 2,
+    };
+    const newPosX = currentPos.x + (targetPos.x - currentPos.x) * 0.1;
+    const newPosY = currentPos.y + (targetPos.y - currentPos.y) * 0.1;
+
+    appRef.value.stage.position.set(
+      Math.min(0, Math.max(newPosX, appRef.value.screen.width - appRef.value.screen.width * appRef.value.stage.scale.x)),
+      Math.min(0, Math.max(newPosY, appRef.value.screen.height - appRef.value.screen.height * appRef.value.stage.scale.y)),
+    );
+  }
```

Comment:
Why do we need to check for `username` here?

```suggestion
  const currentPos = appRef.value.stage.position;
  const targetPos = {
    x: -botx * appRef.value.stage.scale.x + appRef.value.screen.width / 2,
    y: -boty * appRef.value.stage.scale.y + appRef.value.screen.height / 2,
  };
  const newPosX = currentPos.x + (targetPos.x - currentPos.x) * 0.1;
  const newPosY = currentPos.y + (targetPos.y - currentPos.y) * 0.1;

  appRef.value.stage.position.set(
    Math.min(0, Math.max(newPosX, appRef.value.screen.width - appRef.value.screen.width * appRef.value.stage.scale.x)),
    Math.min(0, Math.max(newPosY, appRef.value.screen.height - appRef.value.screen.height * appRef.value.stage.scale.y)),
  );
```

Reply:
Its so that the bot that is assigned to the account is followed when the button is pressed.
I guess the username check can be moved to when the function is being called.

Reply:
> I guess the username check can be moved to when the function is being called.

Yes, this would make more sense. Thank you!